MozillaFoundation/foundation.mozilla.org
# created campaign index featured page relationship

# Description



Link to sample test page: https://foundation-s-featured-c-bfvssc.herokuapp.com/en/campaigns/
Related PRs/issues: https://mozilla-hub.atlassian.net/browse/TP1-27

This pull request introduces a new `featured_page` section to the CampaignIndexPage model. This enhancement enables users to manually curate the list of pages rendered on the campaign index page at `/campaigns/`.

Additionally, this PR includes updates to tests and factories to support this new feature.

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-397)


## Steps to test:
1. Visit the CMS edit page at [this link](https://foundation-s-featured-c-bfvssc.herokuapp.com/cms/pages/93/edit/) using the credentials: **admin2/admin2**
2. Make a change to the "featured pages" section (for example, reorder the list or delete/add an item)
3. Confirm that your changes appear correctly in the preview panel
4. Publish your changes
5. Visit https://foundation-s-featured-c-bfvssc.herokuapp.com/en/campaigns/
6. The list of pages shown here should match what you set in the CMS
7. Visit https://foundation-s-featured-c-bfvssc.herokuapp.com/fr/campaigns/
8. Note that the page "Trade tough once" is now showing its localized version "Trade tough once(in french)"
9. If the campaign page is working as expected (load more, etc). Testing is complete!
